modules/Bio/EnsEMBL/Hive/HivePipeline.pm

```diff
@@ -700,58 +759,71 @@ sub apply_tweaks {
                                 foreach my $df_target ( @{$df_rule->get_my_targets} ) {
                                     $dft_collection->forget_and_mark_for_deletion( $df_target );
 
-                                    print "Tweak.Deleting\t".$df_target->toString." --> (missing value)\n";
+                                    push @response, "Tweak.Deleting\t".$df_target->toString." --> (missing value)\n";
                                 }
                                 $dfr_collection->forget_and_mark_for_deletion( $df_rule );
                                 $need_write = 1;
 
-                                print "Tweak.Deleting\t".$df_rule->toString." --> (missing value)\n";
+                                push @response, "Tweak.Deleting\t".$df_rule->toString." --> (missing value)\n";
                             }
                         }
                     }
 
                     if($operator eq '=' or $operator eq '+=') {     # create new rules
+                        $tweakStructure->{Action} = 'SET';
+                        $tweakStructure->{Return}->{NewValue} = $new_value;
```

Comment:
`$new_value` is the only thing we have but since we havne't defined the format for dataflow rules, should it be skipped ?

Reply:
We aimed to have the same son structure for all responses so there should be serious reasons to break it.